### ISSUE TITLE
Diagnostic Settings schema

### ISSUE THREAD

**Issue by jrspinella** [Azure/azure-resource-manager-schemas#2708]
Bicep version
v0.13.1

AZCLI version
v2.43.0

It doesn't look like diagnosticSettings is supported in schemas for the available types.

Using Schema "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json#"

Errors from Deployments:

```
{
  "code": "DeploymentFailed",
  "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
  "details": [
    {
      "code": "InvalidTemplate",
      "message": "Deployment template parse failed: 'Required property 'type' not found in JSON. Path ''.'."
    }
  ]
} 
```

**Additional context**
![image](https://user-images.githubusercontent.com/5787207/206772559-508a37e4-01c9-4a36-9eac-f31598d85196.png)

![image](https://user-images.githubusercontent.com/5787207/206772062-e5c20414-b863-4d09-8cce-eb2bac8dd551.png)

**Comment by MercyMarkus** [comment]
I can confirm this as well. I've been blocked on this for a week now and tried to look for workarounds.

Came to create an issue and saw that you already did.

**Comment by MercyMarkus** [comment]
**Update**: I've been able to deploy my ARM template successfully despite this warning #https://github.com/Azure/bicep/issues/9247#issuecomment-1356769542

**Comment by jrspinella** [comment]
This has been fixed. https://github.com/Azure/NoOpsAccelerator/issues/226

**Comment by TiepiNL** [comment]
> **Update**: I've been able to deploy my ARM template successfully despite this warning #[Azure/bicep#9247 (comment)](https://github.com/Azure/bicep/issues/9247#issuecomment-1356769542)

I receive the same warning:
![image](https://github.com/Azure/azure-resource-manager-schemas/assets/7967223/9dd9da07-c780-4530-9dc3-1e3e85666f7b)

Is resolving this within the scope of this issue, since the failing deployment already has been fixed?
